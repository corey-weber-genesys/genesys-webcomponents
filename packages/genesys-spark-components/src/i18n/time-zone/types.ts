import { timeZoneIdentifiers } from './identifiers';

export type GuxTimeZoneIdentifier = (typeof timeZoneIdentifiers)[number];
