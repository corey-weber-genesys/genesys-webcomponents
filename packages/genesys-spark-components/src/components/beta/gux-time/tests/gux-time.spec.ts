import { newSpecPage } from '@stencil/core/testing';
import { GuxTime } from '../gux-time';
import { timeZoneIdentifiers } from '../../../../i18n/time-zone/identifiers';

const components = [GuxTime];
const language = 'en';
const date = '2022-07-07T13:35:30.100Z';

describe('gux-time', () => {
  describe('#render different time formats eg(short,medium,full,long)', () => {
    [
      `<gux-time-beta datetime=${date} format="short"></gux-time-beta>`,
      `<gux-time-beta datetime=${date} format="medium"></gux-time-beta>`,
      `<gux-time-beta datetime=${date} format="full"></gux-time-beta>`,
      `<gux-time-beta datetime=${date} format="long"></gux-time-beta>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('#render different time zones', () => {
    timeZoneIdentifiers.forEach((timeZone: string) => {
      it(`should work as expected for "${timeZone}"`, async () => {
        const html = `<gux-time-beta time-zone=${timeZone} datetime="2022-07-07T12:00:00.000Z" format="full"></gux-time-beta>`;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
