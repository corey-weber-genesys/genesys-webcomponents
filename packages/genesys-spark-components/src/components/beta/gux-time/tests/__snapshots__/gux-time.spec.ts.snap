// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (1) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="short">
  <mock:shadow-root>
    1:35 PM
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (2) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="medium">
  <mock:shadow-root>
    1:35:30 PM
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (3) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="full">
  <mock:shadow-root>
    1:35:30 PM Coordinated Universal Time
  </mock:shadow-root>
</gux-time-beta>
`;

exports[`#render different time formats eg(short,medium,full,long) should render component as expected (4) 1`] = `
<gux-time-beta datetime="2022-07-07T13:35:30.100Z" format="long">
  <mock:shadow-root>
    1:35:30 PM UTC
  </mock:shadow-root>
</gux-time-beta>
`;
